judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/AutoMappedTransferObjectTest.java

```diff
@@ -0,0 +1,127 @@
+import hu.blackbelt.judo.psm.generator.sdk.core.test.api.automappedtransfermodel.automappedtransfermodel.automappedtransferonabstractoptionalfields.AutoMappedTransferOnAbstractOptionalFields;
+import hu.blackbelt.judo.psm.generator.sdk.core.test.api.automappedtransfermodel.automappedtransfermodel.automappedtransferonabstractoptionalfields.AutoMappedTransferOnAbstractOptionalFieldsDao;
+import hu.blackbelt.judo.psm.generator.sdk.core.test.api.automappedtransfermodel.automappedtransfermodel.automappedtransferonoptionalfields.AutoMappedTransferOnOptionalFields;
+import hu.blackbelt.judo.psm.generator.sdk.core.test.api.automappedtransfermodel.automappedtransfermodel.automappedtransferonoptionalfields.AutoMappedTransferOnOptionalFieldsDao;
+import hu.blackbelt.judo.psm.generator.sdk.core.test.api.automappedtransfermodel.automappedtransfermodel.automappedtransferonrequiredfields.AutoMappedTransferOnRequiredFields;
+import hu.blackbelt.judo.psm.generator.sdk.core.test.api.automappedtransfermodel.automappedtransfermodel.automappedtransferonrequiredfields.AutoMappedTransferOnRequiredFieldsDao;
+import hu.blackbelt.judo.psm.generator.sdk.core.test.guice.AutoMappedTransferModelDaoModules;
+import hu.blackbelt.judo.requirement.report.annotation.Requirement;
+import hu.blackbelt.judo.requirement.report.annotation.TestCase;
+import hu.blackbelt.judo.runtime.core.jsl.AbstractJslTest;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Slf4j
+public class AutoMappedTransferObjectTest extends AbstractJslTest {
+
+    @Override
+    public Module getModelDaoModule() {
+        return new AutoMappedTransferModelDaoModules();
+    }
+
+    @Override
+    public String getModelName() {
+        return "AutoMappedTransferModel";
+    }
+
+    @Inject
+    AutoMappedTransferOnOptionalFieldsDao autoMappedTransferOnOptionalFieldsDao;
+
+    @Inject
+    AutoMappedTransferOnRequiredFieldsDao autoMappedTransferOnRequiredFieldsDao;
+
+    @Inject
+    AutoMappedTransferOnAbstractOptionalFieldsDao autoMappedTransferOnAbstractOptionalFieldsDao;
+
+    @Inject
+    AbstractEntityFieldsDao abstractEntityFieldsDao;
+
+    /**
+     * [DESCRIPTION]
+     *
+     * @prerequisites The model runtime is empty. It means that the database of the application has to be empty before this test starts running.
+     *
+     * @type Behaviour
+     *
+     * @others Implement this test case in the *judo-runtime-core-jsl-itest* module.
+     *
+     * @jslModel AutoMappedTransferModel.jsl
+     * model AutoMappedTransferModel;
+     *
+     *
+     * @positiveRequirements
+     *
+     * @negativeRequirements
+     *
+     * @scenario
+     *
+     */
+    @Test
+    @TestCase("AutoMappedTransferObjectFields")
+    @Requirement(reqs = {
+            "",
+    })
+    void testCheckOfTransferAutoMappedTransferObjectsCreationWithValidFields() {
```

Comment:
For testing mapping you could create an instance "through" a transfer then query that same instance as an entity and check its values. The main goal is that to test each (auto)mapped attribute/reference is actually mapped

Reply:
This is under development yet, I will implement the automapped test later.